### ISSUE TITLE
no-extra-boolean-cast - extend for Boolean()

### ISSUE THREAD

**Issue by czjvic** [eslint/eslint#3650]
This rule should probably report also following construction:
```javascript
if (Boolean(variable)) {
}
```

Version 1.3.1

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by ilyavolodin** [comment]
I think I agree. @eslint/eslint-team any thoughts?

**Comment by nzakas** [comment]
:+1:

**Comment by platinumazure** [comment]
Just out of curiosity, how exactly would the rule deal with the possibility of someone redefining `Boolean`? Or aliasing something else to the same constructor?

Maybe the answer is "ESLint is not a semantic analyzer that tracks the value of variables", but... seemed worth asking.

**Comment by IanVS** [comment]
@platinumazure I believe that would be covered by [no-native-reassign](http://eslint.org/docs/rules/no-native-reassign).

**Comment by nzakas** [comment]
@platinumazure ESLint is not a semantic analyzer that tracks the value of variables :)

**Comment by platinumazure** [comment]
@nzakas Har har har. :-)
